### ISSUE TITLE
Support `editable: false` for workspace dependencies

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/uv#5988]
See: #5985.

**Comment by charliermarsh** [comment]
I'm wondering if this should be a property of the workspace member itself rather than a property of the _dependency_?

**Comment by charliermarsh** [comment]
The more I think about this, the less I like it. Even if we support `editable: false`, the _requiring_ package will still be installed as editable. I think editable should either be (1) a property of the package (not the dependency), or better (2) a global setting.

**Comment by zanieb** [comment]
I don't think we need to make a decision on this for the upcoming stabilizations, but don't have strong feelings about how it works. Curious to hear from @konstin and community members trying out workspaces.

**Comment by charliermarsh** [comment]
Agree. Just hard for me to imagine a use-case where `root` depends on `child`, and you want `child` installed as non-editable but `root` as editable? And then there's _no_ way to install `root` as non-editable?

**Comment by LucidDan** [comment]
I have one scenario I ran into that led me to this issue.

If I'm building a mono-repo using uv workspaces, and I have a type stubs package that I want to have in my repo as a workspace that is used by several projects...mypy doesn't work properly with editable dependencies (see https://github.com/python/mypy/issues/12313, apparently there is no plan to change this behaviour)

In that case, I need the type stubs package installed non-editable, even when the rest of the workspaces are editable.
I work around this by defining the type stub as a package instead of a workspace source, eg:

```
[tool.uv.sources]
demo = { workspace = true }
lh = { workspace = true }
my-typestubs = { path = "packages/my-typestubs", editable = false }
```

This is a reasonable workaround - it just means I end up with a separate lock and .venv inside the type stubs package, and its slightly slower than if I could keep it as a workspace.

I can't really judge if this is something many people would want to do, but for me at least, it's a valid use case.

**Comment by zanieb** [comment]
Thanks for weighing in! cc @AlexWaygood

**Comment by xmakro** [comment]
We are also running into this use case in a large monorepo.

```
[tool.uv.sources]
demo = { workspace = true, editable = false }
```

would be ideal

**Comment by hongquan** [comment]
I run into the same situation, my repo includes a `logbook-stubs` (the 3rd-party `logbook` is not typed yet, so I want to create stubs and test first, before contributing upstream) as workspace member, and `ty` doesn't resolve the type import from this package if it is installed as editable.

**Comment by sharifhsn** [comment]
Closed by #15708
